blog/2020-01-angular-scully.md

```diff
@@ -0,0 +1,332 @@
+To visit this route and pre-render it, we need a way to fill the `isbn` parameter.
+Luckily _Scully_ helps us with this too.
+We can configure a — so called [_Route Plugin_](https://github.com/scullyio/scully/blob/master/docs/plugins.md) — that will call an API, fetch the data from it and pluck the `isbn` from the array of results to fill it in the router parameter.
+
+In the following example we will use [BookMonkey API](https://api3.angular-buch.com) as an API to fetch a list of books:
+
+```js
+exports.config = {
+  projectRoot: "./src/app",
+  routes: {
+    ...
+    '/books/:isbn': {
+      'type': 'json',
+      'isbn': {
+        'url': 'https://api3.angular-buch.com/books',
+        'property': 'isbn'
+      }
+    }
+  }
+};
+```
+
+The result from the API call will look like the following:
+
+```json
+[
+  {
+    "title": "Angular",
+    "subtitle": "Grundlagen, fortgeschrittene Themen und Best Practices – mit NativeScript und NgRx",
+    "isbn": "9783864906466",
+    ...
+  },
+  {
+    "title": "Angular",
+    "subtitle": "Grundlagen, fortgeschrittene Techniken und Best Practices mit TypeScript - ab Angular 4, inklusive NativeScript und Redux",
+    "isbn": "9783864903571",
+    ...
+  },
+  ...
+]
+```
+
+After _Scully_ plucks the ISBN, it will just iterate over the final array: `['9783864906466', '9783864903571']`.
+In fact when running _Scully_ using `npm run scully`, it will visit the following routes, **after we have configured the route `/books/:isbn` in the Angular router** (otherwise non used routes will be skipped).
+
+```text
+/books/9783864906466
+/books/9783864903571
+```
+
+We can see the result in the log:
+
+```text
+ ☺   new Angular build imported
+Background servers already running.
+servers available
+Finding all routes in application.
+Pull in data to create additional routes.
+Finding files in folder "<path>/scully-blog/blog"
+Route list created in files:
+      src/assets/scully-routes.json
+      /<path>/scully-blog/dist/static/assets/scully-routes.json
+Route "/books/9783864903571" rendered into file: "/<path>/scully-blog/dist/static/books/9783864903571/index.html"
+Route "/books/9783864906466" rendered into file: "/<path>/scully-blog/dist/static/books/9783864906466/index.html"
+```
+
+If you you wanna follow all the development steps in detail, check out my provided github repository
+[scully-blog-example](https://github.com/d-koppenhagen/scully-blog-example/commits).
+Each step described here is represented by one commit.
+
+## Conclusion
+
+Scully is awesome if you need a pre-rendered Angular SPA where all routes can be accessed immediately without loading the whole app at once.
+This is a great benefit for users as they don’t need to wait until the whole bunch of JavaScript has been downloaded to their devices.
+They have instantly access to the sites information.
+Furthermore _Scully_ offers a way to create very easily a blog and renders all posts written in markdown.
```

Comment:
I'd love to see a reference to traditional pre-rendering here. Except for convenience and for the blog feature Scully doesn't seem to be very different to normal pre-rendering with the Universal tooling. So what makes Scully great seems to be that it's so easy

Reply:
1. It's easier than Universal.
2. With pre-rendering (not universal) you can possibly get away with NOT shipping parts of your backend to production. If the calls are cached at build time, then your app in production can make less calls to the backend. 
3. Universal could never render an Angular hybrid app. Or an Angular app with a jquery plugin. But Scully has no issues with this.

Reply:
There are more. These are a few.